routes/User.js

```diff
@@ -0,0 +1,141 @@
+const express = require("express");
+const router = express.Router();
+const magic = require("../magic");
+const { User, Chapter, Module } = require("../models");
+const { Op } = require("sequelize");
+
+router.get("/", async (req, res) => {
+	const { uuid, xtz } = req.query;
+	console.log(uuid, xtz);
+
+	try {
+		const param = uuid ? { uuid } : { xtzAddress: xtz };
+
+		const user = await User.findOne({ where: param });
+
+		if (!user) {
+			throw new Error();
+		}
+
+		res.json(user);
+	} catch {
+		res.status(404).json({ error: "USER_NOT_FOUND" });
+	}
+});
+
+router.post("/", async (req, res) => {
+	const { xtzAddress } = req.body;
+
+	try {
+		let user = await User.findOne({ where: { xtzAddress } });
+
+		if (!user) user = await User.create({ xtzAddress });
+
+		return res.json(user);
+	} catch (err) {
+		return res.status(500).json(err.toString());
+	}
+});
+
+router.patch("/", async (req, res) => {
```

Comment:
What's this route being used for?

Reply:
### `/user` [PATCH]
First when the user is created, `name` and `email` field are null.
This route updates a user with the name and email provided, if, a user isn't already using the same name or email.

Reply:
Got it.